File: python-seq-codes/spins.py
import numpy as np
import para
class Spins:
    def __init__(self) -> None:
        self.N = None
        self.M = None
        self.A = []

    # def init(self, N: int, M: int):
    #     self.N = N
    #     self.M = M
    #     self.A = 
    def energy(self):
        return (-np.sum(self.A[1:para.N+1,1:para.M+1]*self.A[0:para.N,1:para.M+1]  \
            +  self.A[1:para.N+1,1:para.M+1]*self.A[2:para.N+2,1:para.M+1] \
            +  self.A[1:para.N+1,1:para.M+1]*self.A[1:para.N+1,0:para.M] \
            +  self.A[1:para.N+1,1:para.M+1]*self.A[1:para.N+1,2:para.M+2]))

File: julia-seq-code/src/spins.jl
import para

struct Spins
	N::Int64
	M::Int64
	A::Array
end

energy(S::Spins) = sum(
+(S.A[2:para.N+2,2:para.M+2] * S.A[1:para.N+1,2:para.M+2]
, S.A[2:para.N+2,2:para.M+2] * S.A[3:para.N+3,2:para.M+2]
, S.A[2:para.N+2,2:para.M+2] * S.A[2:para.N+2,1:para.M+1]
, S.A[2:para.N+2,2:para.M+2] * S.A[2:para.N+2,3:para.M+3]))
